otterdog/eclipse-dataspacetck.jsonnet

```diff
@@ -11,8 +11,21 @@ local newRepo(repoName) = orgs.newRepo(repoName) {
       has_wiki: false,
       allow_auto_merge: true,
       dependabot_security_updates_enabled: true,
+      gh_pages_build_type: "legacy",
+      gh_pages_source_branch: "gh-pages",
+      gh_pages_source_path: "/",
+      squash_merge_commit_message: "PR_TITLE",
+      environments: [
+        orgs.newEnvironment('github-pages') {
+            branch_policies+: [
+                "gh-pages"
+            ],
+            deployment_branch_policy: "selected"
+        }
+      ],
       branch_protection_rules: [
         orgs.newBranchProtectionRule('main') {
+          requires_pull_request: false, //sometimes maintainers need to push to main
```

Comment:
so if you have the need to directly push to the branch without a pull request, this can not be modelled with branch protection rules when status checks are also enabled.

status checks are mandatory in branch protection rules, and even bypass rules do not help.
For cases like that rulesets come to the rescue, take a look here:

https://github.com/eclipse-csi/.eclipsefdn/blob/main/otterdog/eclipse-csi.jsonnet#L5-L7

We setup something like this for several projects already.

Reply:
ok, do you have an easy way to remove the `require_pull_request: false` line?

Reply:
I can fix that, need to fix the squash setting as well. Will add a validation check to ensure only valid combinations are selected.

Reply:
/otterdog done

Reply:
ok I have now also added some more validations for squash and merge commit settings: https://github.com/eclipse-csi/otterdog/commit/7cba39a928184a9b5352e1fd5302ff020c245529

So in the future, you will be warned if you try to set something that is not supported by GitHub.